src/main/java/az/edu/turing/module02/part01/lesson02/LinkedList.java

```diff
@@ -0,0 +1,192 @@
+package az.edu.turing.module02.part01.lesson02;
+
+import java.lang.reflect.Array;
+import java.util.NoSuchElementException;
+
+public class LinkedList<T> {
+    LinkNode<T> head;
+    int size = 0;
+
+    public void addHead(T item) {
+        LinkNode<T> itemNode = new LinkNode<>(item);
+        itemNode.next = head;
+        head = itemNode;
+        size++;
+    }
+
+    public void addTail(T item) {
+        LinkNode<T> itemNode = new LinkNode<>(item);
+
+        if (head == null) {
+            head = itemNode;
+            size++;
+            return;
+        }
+
+        LinkNode<T> lastNode = getLastNode();
+        lastNode.next = itemNode;
+        size++;
+    }
+
+    public void removeHead() {
+        if (head == null) throw new NoSuchElementException();
+
+        head = head.next;
+        size--;
+    }
+
+    public void removeTail() {
+        if (head == null) throw new NoSuchElementException();
+
+        if (size == 1) {
+            head = null;
+            size--;
+            return;
+        }
+
+        LinkNode<T> nodeIterator = head;
+
+        for (int i = 0; i < size - 2; i++) {
+            nodeIterator = nodeIterator.next;
+        }
+
+        nodeIterator.next = null;
+        size--;
+    }
+
+    public void insert(int index, T item) {
+        if (index < 0 || index >= size) throw new IndexOutOfBoundsException();
+
+        if (index == 0) {
+            addHead(item);
+            return;
+        }
```

Comment:
can be added one more check for if index at the end of the list, then we could call addTail method